kore/src/Kore/Rewrite/Axiom/Matcher.hs

```diff
@@ -393,12 +394,69 @@ patternMatch' sideCondition ((MatchItem pat subject boundVars boundSet) : rest)
             decomposeBinder (inject variable1) term1 (inject variable2) term2
         (Exists_ _ variable1 term1, Exists_ _ variable2 term2) ->
             decomposeBinder (inject variable1) term1 (inject variable2) term2
-        (App_ symbol1 children1, App_ symbol2 children2) ->
-            if symbol1 == symbol2
-                then decomposeList (zip children1 children2)
-                else
-                    failMatch $
-                        "distinct application symbols: " <> (unparseToText symbol1) <> ", " <> (unparseToText symbol2)
+        (_, _)
+            | Just False <- List.isListSort tools sort
+            , (App_ symbol1 children1, App_ symbol2 children2) <- (pat, subject) ->
```

Comment:
The top-level symbol application case now excludes the symbols of sort `List`. There's a duplicate of this case withing the `List` matching below.